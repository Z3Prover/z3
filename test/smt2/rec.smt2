(define-fun-rec f ((x Int) (y Bool)) Int
    (if y (f (+ x 1) (not y)) x))

(define-funs-rec 
   ((ping ((x Int) (y Bool)) Int)
    (pong ((a Int) (b Int)) Bool))

   ((if y (ping (+ x 1) (pong x x)) x)
    (if (> a b) (ping a (pong b b)) b)))

(assert (< (f 0 true) 0))
(check-sat)