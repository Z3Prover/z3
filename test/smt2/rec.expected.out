(error "line 9 column 38: invalid function definition, sort mismatch. Expcected Bool but function body has sort Int")
unsat
